src/formats/cif.cpp

```diff
@@ -0,0 +1,205 @@
+        {}
+
+    int find_row(gemmi::cif::Table &table, int start_idx = 0) const {
+        int idx = start_idx; 
+        int total_rows = table.length();
+
+        assert(idx < total_rows);
+
+        while (idx < total_rows) {
+            if (this->is_row(table[idx])) {
+                return idx;
+            }
+            ++idx;
+        }
+
+        fmt::print(stderr, "In MCRA::find_row(), unable to find Atom(1, {}, {}, {}, {})\n"
+                    "Looping through entire table to double check...",
+                    this->_chain, this->_res_num, this->_residue, this->_atom); 
+
+        idx = 0;
+        for (const auto site : table) {
+            if (this->is_row(site)) {
+                return idx;
+            }
+            ++idx;
+        }
+        return -1;
+    }
+
+    friend std::ostream &operator<<(std::ostream &os, const MCRA &mcra) {
+        os << "Atom(" << mcra._model << " " << mcra._chain << " " << mcra._res_num << " [" << mcra._residue << "] " << mcra._atom << ")";
+        return os;
+    }
+
+private:
+    bool is_row(const gemmi::cif::Table::Row &row) const {
+        int model = std::stoul(row[11]);
+        const std::string &chain = row[1];
+        const std::string &residue = row[4];
+        const std::string &res_num = row[2];
+        const std::string atom = row[5][0] != '"' ? row[5] : row[5].substr(1, row[5].size() - 2);
+
+        if (_model == model) {
+            if (_chain == chain) {
+                if (_res_num == res_num && _residue == residue) {
+                    if (_atom == atom) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+};
+
+
+void CIF::write_cif_block(std::ostream &out,
+                          gemmi::cif::Table &table, 
+                          std::vector<std::string> &p_charge, 
+                          std::vector<std::string> &vdw_radii) {
+
+    auto &loop = *table.get_loop();
+
+    unsigned long orig_tag_size = loop.tags.size();
+    unsigned long new_tag_size = orig_tag_size + 2;
+
+    // Creates a new table full of empty strings with the correct number of dimensions
+    // Outside vector size is the # of columns, inside vector size is the # of rows.
+    std::vector<std::vector<std::string>> newCols(new_tag_size, {loop.length(), {"Empty"}});
+
+    // Copies data from original columns to their respecitve column in the new table filled with empty strings.
+    // Leaving only the new appended columns as empty strings
+    for (unsigned int i = 0; i != orig_tag_size; ++i) {
+        auto iCol = table.bloc.find_loop(loop.tags[i]);
+        std::copy(iCol.begin(), iCol.end(), newCols[i].begin());
+    }
+
+    newCols[new_tag_size - 2] = std::move(p_charge);
+    newCols[new_tag_size - 1] = std::move(vdw_radii);
+
+    std::vector<std::string> new_tags{
+        "_atom_site.fw2_charge",
+        "_atom_site.fw2_vdw_radius"};
```

Comment:
Do we need the radius? I know PQR writer has that, but the goal was just to comply with "definitions" here:
https://www.ics.uci.edu/~dock/manuals/apbs/html/user-guide/a2566.html
or here: https://docs.mdanalysis.org/1.0.1/documentation_pages/coordinates/PQR.html

Do you need that for some application in the cif?

Reply:
I personally do not use the radius. This is was inherited from my implementation of this feature in[ freesasa](https://github.com/mittinatten/freesasa/blob/master/src/cif.cc). 

I am unsure if anyone currently needs it for a cif application, its more for being consistent and enabling people to move to cif in the future where they currently use pqr. 

Let me know if you want to remove it. In my opinion, I think its best to leave it for other people whom might need it.